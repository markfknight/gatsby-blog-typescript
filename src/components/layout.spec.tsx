import React, { PropsWithChildren } from 'react';
import renderer from 'react-test-renderer';

import Layout, { LayoutProps } from './layout';

describe(`Layout snapshot`, () => {
  it(`renders correctly when root pathname`, () => {
    const props = {
      title: `test`,
      location: {
        pathname: `/`,
      },
    } as PropsWithChildren<LayoutProps>;

    const tree = renderer
      .create(
        <Layout {...props}>
          <p>Test</p>
        </Layout>
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  it(`renders correctly when not root pathname`, () => {
    const props = {
      title: `test`,
      location: {
        pathname: ``,
      },
    } as PropsWithChildren<LayoutProps>;

    const tree = renderer
      .create(
        <Layout {...props}>
          <p>Test</p>
        </Layout>
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
