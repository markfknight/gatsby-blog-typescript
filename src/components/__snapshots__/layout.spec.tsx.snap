// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Layout snapshot renders correctly when not root pathname 1`] = `
<div>
  <header>
    <h3>
      <a
        href="/"
        style={
          Object {
            "boxShadow": "none",
            "color": "inherit",
            "textDecoration": "none",
          }
        }
      >
        test
      </a>
    </h3>
  </header>
  <main>
    <p>
      Test
    </p>
  </main>
  <footer>
    © 
    2021
    , Built with
     
    <a
      href="https://www.gatsbyjs.org"
    >
      Gatsby
    </a>
  </footer>
</div>
`;

exports[`Layout snapshot renders correctly when root pathname 1`] = `
<div>
  <header>
    <h1>
      <a
        href="/"
        style={
          Object {
            "boxShadow": "none",
            "color": "inherit",
            "textDecoration": "none",
          }
        }
      >
        test
      </a>
    </h1>
  </header>
  <main>
    <p>
      Test
    </p>
  </main>
  <footer>
    © 
    2021
    , Built with
     
    <a
      href="https://www.gatsbyjs.org"
    >
      Gatsby
    </a>
  </footer>
</div>
`;
